src/Symfony/Component/AssetMapper/Command/ImportMapRequireCommand.php

```diff
@@ -110,28 +127,45 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             );
         }
 
-        $newPackages = $this->importMapManager->require($packages);
+        if ($input->getOption('dry-run')) {
+            $newPackages = $this->importMapManager->requirePackages($packages, new ImportMapEntries());
+        } else {
+            $newPackages = $this->importMapManager->require($packages);
+        }
 
         $this->renderVersionProblems($this->importMapVersionChecker, $output);
 
-        if (1 === \count($newPackages)) {
-            $newPackage = $newPackages[0];
-            $message = \sprintf('Package "%s" added to importmap.php', $newPackage->importName);
+        $newPackageNames = array_map(fn (ImportMapEntry $package): string => $package->importName, $newPackages);
 
-            $message .= '.';
+        if (1 === \count($newPackages)) {
+            $messages = [\sprintf('Package "%s" added to importmap.php.', $newPackageNames[0])];
         } else {
-            $names = array_map(fn (ImportMapEntry $package) => $package->importName, $newPackages);
-            $message = \sprintf('%d new items (%s) added to the importmap.php!', \count($newPackages), implode(', ', $names));
+            $messages = [\sprintf('%d new items (%s) added to the importmap.php!', \count($newPackages), implode(', ', $newPackageNames))];
         }
 
-        $messages = [$message];
+        if ($io->isVerbose()) {
+            $io->table(
+                ['Package', 'Version', 'Path'],
+                array_map(fn (ImportMapEntry $package): array => [
+                    $package->importName,
+                    $package->version ?? '-',
+                    // BC layer for FrameworkBundle < 7.3
+                    // When projectDir is not set, we use the absolute path of the package
+                    isset($this->projectDir) ? Path::makeRelative($package->path, $this->projectDir) : $package->path,
```

Comment:
I suggest using a `null` check instead of a `isset` check. This way, it will make it clear that the check becomes irrelevant once the property is not nullable anymore (`isset` could still be relevant because it also checks for uninitialized properties, requiring more analysis).

Reply:
Done, thanks !